File: FangZhiHu/app/src/main/java/network/exception/ServiceException.java
package network.exception;

/**
 * Created by yuantongqin on 2017/8/15.
 * modify by liyanxi on 2017/9/27
 * todo 新增参数data，考虑到请求状态不为0时，后端也有信息返回需要根据实际情况解析
 */

public class ServiceException extends RuntimeException {
    //服务器运行时异常
    private int code;
    public int customState;
    private String message;
    private String data;

    public ServiceException(int code, String msg, String data) {
        this.code = code;
        this.message = msg;
        this.data = data;
    }

    public ServiceException(int code, String data) {
        this.code = code;
        this.data = data;
    }

    public ServiceException(int customState,int code, String data) {
        this.customState = customState;
        this.code = code;
        this.data = data;
    }

    public int getCode() {
        return code;
    }

    @Override
    public String getMessage() {
        return message;
    }

    public String getData() {
        return data;
    }


}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/excepition/ServiceException.kt
package network.excepition

import java.lang.RuntimeException

/**
 * Created by tongqinyuan on 2018/3/7.
 */
class ServiceException :RuntimeException {
     var code: Int = 0
        private set
    var customState: Int = 0

     var messages: String? = null
        private set

     var data: String? = null
        private set


    constructor(code:Int,msg:String,data:String){
        this.code = code
        this.messages = msg
        this.data = data
    }
    constructor(code:Int,data:String){
        this.code = code
        this.data = data
    }
    constructor(code:Int,customState:Int,data:String){
        this.customState = customState
        this.code = code
        this.data = data
    }
}